tools/run_tests/run_tests.py

```diff
@@ -1132,6 +1132,8 @@ def __str__(self):
     'node': NodeLanguage(),
     'node_express': NodeExpressLanguage(),
     'php': PhpLanguage(),
+    'php_protobuf_php': PhpLanguage(),
```

Comment:
I just realized this is in run_tests.py.
1. I think there should be better ways around this than adding 2 more languages as a "hack".
2. It sounds like 'php_protobuf_php'  is the same thing as 'php' (as php protobuf is the default).
3. If you have a run_tests.py language 'php_protobuf_c', it should really build php with protobuf C extension, but the definition for php_protobuf_php and php_protobuf_c seem identical - which is pretty confusing and possibly wrong.

I don't understand why adding the "virtual" languages to run_tests.py actually prevents a conflict in 
https://github.com/grpc/grpc/blob/master/tools/run_tests/performance/run_worker_php.sh
(btw, I also noticed you are running some steps that should belong to build_performance.sh as part of "run worker" script, which doesn't seem correct -   phpize, composer install, make, configure are clearly build steps).

Also, php_protobuf_php is the "default" setting, so you should only have

Reply:
I revert the changes in run_tests.py.
The reason I modify there is `workers_by_lang = dict([(str(language), []) for language in languages])` in run_performance.py. `php` and `php_protobuf_c` shares the same string method, thus when create workers, it doubled what I need. Thus I want to use different name to distinguish them. However that name should be accepted by run_tests.py. 
Because `php` and `php_protobuf_c` can share the same build script, thus I define a string method "php_with_argv_xxxx", and if it has "_with_argv" in it, it only need to build once as the language.

Reply:
For the "run worker" script, "composer install" installs the library only used in qps test "client.php", and it actually installs tons of things (clone the whole repo of protobuf and grpc), which is unnecessary if no one runs "run_worker". Thus I put it in this file.
Also "phpize, make" is cd to the protobuf repo downloaded by "composer install" and compile a protobuf.so c-extension, which is only needed by the qps test.

Reply:
So there's 3 stages:
1. php built by run_tests.py -l php --build_only  (here I agree you don't need to build qps test stuff - even though some languages actually do so)

2. php qps worker built by build_performance.sh script - this is where you should build everything that's required for running php qps tests (all other languages do so).

3. run_php_workers.sh  - this should really contain just commands to run an already-built php qps worker. If you include build steps here, building runs a part of qps worker invocation so there's going to be a delay (possibly a long one) before the qps worker is started - right now the qps_driver happens to work (because unlimited retries) with such delay, but it's more of a coincidence rather than something we are doing consciously.

Please look at
https://github.com/grpc/grpc/blob/master/tools/run_tests/performance/build_performance.sh#L44
Idea:
you can actually provide case statement for both "php" and "php_protobuf_c" here and easily run the right build commands. That would save you from doing all the "magic" in run_performance_tests.py  (unifying php and php_with_argv etc.).

Reply:
Thanks for reverting the php language proliferation in run_tests.py btw.